pkg/controller/controller_binding.go

```diff
@@ -216,36 +234,46 @@ func (c *controller) reconcileBinding(binding *v1alpha1.Binding) error {
 		if err != nil {
 			s := fmt.Sprintf("Error creating Binding \"%s/%s\" for Instance \"%s/%s\" of ServiceClass %q at Broker %q: %s", binding.Name, binding.Namespace, instance.Namespace, instance.Name, serviceClass.Name, brokerName, err)
 			glog.Warning(s)
-			c.updateBindingCondition(
+			if err := c.updateBindingCondition(
 				binding,
 				v1alpha1.BindingConditionReady,
 				v1alpha1.ConditionFalse,
 				errorBindCallReason,
-				"Bind call failed. "+s)
+				"Bind call failed. "+s,
+			); err != nil {
+				glog.Errorf("updating binding condition (%s)", err)
+				return err
+			}
 			c.recorder.Event(binding, api.EventTypeWarning, errorBindCallReason, s)
 			return err
 		}
 		err = c.injectBinding(binding, &response.Credentials)
 		if err != nil {
 			s := fmt.Sprintf("Error injecting binding results for Binding \"%s/%s\": %s", binding.Namespace, binding.Name, err)
 			glog.Warning(s)
-			c.updateBindingCondition(
+			if err := c.updateBindingCondition(
 				binding,
 				v1alpha1.BindingConditionReady,
 				v1alpha1.ConditionFalse,
 				errorInjectingBindResultReason,
 				"Error injecting bind result "+s,
-			)
+			); err != nil {
+				glog.Errorf("updating binding condition (%s)", err)
+				return err
+			}
 			c.recorder.Event(binding, api.EventTypeWarning, errorInjectingBindResultReason, s)
 			return err
 		}
-		c.updateBindingCondition(
+		if err := c.updateBindingCondition(
```

Comment:
I'm finding this style here to be harder to read than (IMHO) simpler:
err := c.update...
if err != nil {


But, if others are fine with this style, then so be it :)